public/app/features/datasources/components/picker/DataSourceModal.tsx

```diff
@@ -194,11 +196,11 @@ function getDataSourceModalStyles(theme: GrafanaTheme2) {
 
       ${theme.breakpoints.down('md')} {
         width: 100%;
-        height: 47%;
         border-right: 0;
         padding-right: 0;
-        border-bottom: 1px solid ${theme.colors.border.weak};
-        padding-bottom: ${theme.spacing(4)};
+        flex: 1;
+        // This is a hack to make the height of the list work on mobile
+        height: calc(100% - 104px);
```

Comment:
is the problem that it takes too much height? 
what about doing this? 
```suggestion
        flex: 2;
        overflow-y: auto
```